resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -122,8 +122,9 @@ private[spark] class MesosClusterScheduler(
     conf: SparkConf)
   extends Scheduler with MesosSchedulerUtils {
   var frameworkUrl: String = _
+  private val metricsSource = new MesosClusterSchedulerSource(this)
```

Comment:
Shouldn't this be configurable? Is it ok to start the dispatcher by default with a metric system set up?

Reply:
This is just moving the prior initialization from L308 slightly earlier. This is just being done to make the `sourceName` member available below (avoid duplicate `mesos_cluster` constants across different files).